### ISSUE TITLE
OHEA & OIHM - The apps are not Accessible

### ISSUE THREAD

**Issue by daiClegg** [dbrownukk/EFD_v2#619]
The IDAPs service was (briefly) tested for accessibility by UoR IT services.
They are obliged to provide accessible software in their services, of which IDAPs, strictly speaking, is.
Here's the brief summary by the accessibility specialist:

As a result of basic automated (browser Plugin Wave) and manual tests (keyboard, screenreader) on desktop with Windows and chrome. 

I could see some issues that can be easily addressed and could improve the accessibility significantly.
 
At this stage I would only highlight the high priority issues that might have an impact on the usability/accessibility of the platform:  
 
**Language missing or invalid (Site language)** - affected all pages, including login page.
The language of the document is not identified or a lang attribute value is invalid.
 
**Missing form label (User and Password form field)** – same issue on the rest of the form fields that are present on the website
A form control does not have a corresponding label.
 
**Empty button – apply to all confirmation buttons**, such as Sign in, Save, Run buttons
A button is empty or has no value text.
 
**Missing error notification (failed login)** – apply to all error notifications                          
Screenreader users are not aware of the error message.
 
**Keyboard focus is not visible**
Keyboard users might have difficulties to navigate through the page.
 
**Empty table header** – affected all tables
A <th> (table header) contains no text.
The <th> element helps associate table cells with the correct row/column headers. A <th> that contains no text may result in cells with missing or incorrect header information.
 
**Empty link (Sign out link)**
A link contains no text.
If a link contains no text, the function or purpose of the link will not be presented to the user. This can introduce confusion for keyboard and screen reader users.

**Comment by dbrownukk** [comment]
I asked Javier (OpenXava) to comment on Accessibility features of OpenXava - his reply

_We have not done any specific work to make OpenXava applications accessibly. I think it should interesing to add accessibility to OpenXava, indeed some OX developers asked me about adding WAI-ARIA support. Does it will be enough for you?

Certainly if you are creating apps for the public sector you may be obliged (and should) at least consider accessibility.

If you look a the success stories of OpenXava:
https://openxava.org/stories/
Many of them are from public sector. In fact, I work for the public sector using OpenXava applications with no problems. Moreover many of the applications I see in the public sector use RPG in AS/400 with green screens, today in 2021. Are RPG applications accessible?

Anyways, I am in favor of adding accessibility to OpenXava._
